spec/lib/reports/sales_tax_totals_by_order_spec.rb

```diff
@@ -145,7 +145,7 @@
         total = report.total_excl_tax(query_row)
 
         # discounted order total - discounted order tax
-        expect(total).to eq((113.3 - 10) - (3.3 - 0.29))
+        expect(total).to eq((BigDecimal('113.3') - 10) - (BigDecimal('3.3') - BigDecimal('0.29')))
```

Comment:
This looks like a rounding issue, we could just use `round` here :
```suggestion
        expect(total).to eq(((113.3 - 10) - (3.3 - 0.29)).round(2))
```
Curious to see what other @openfoodfoundation/developers think ?

Reply:
Hmm, from what I understand, using BigDecimal here ensures we calculate the exact right figure. 
I guess using floats resulted in something almost quite right, which is why `round` helped snap it to the right number. 

But I must be missing something; I don't understand why upgrading BigDecimal would change the behaviour of floats.

In any case, I'm fine with whatever method is readable and works well enough to ensure the test passes/fails correctly.

Reply:
BTW I think `'113.3'.to_d` is a shortcut for BigDecimal. And we can lose the outer parentheses on `eq`.

Reply:
> from what I understand, using BigDecimal here ensures we calculate the exact right figure.
> ...I don't understand why upgrading BigDecimal would change the behaviour of floats.

It doesn't change behaviour of floats alone but it can change calculations with floats. While the changed right side was all floats, the left side of the equation, the `total`, is calculated with BigDecimal:

https://github.com/openfoodfoundation/openfoodnetwork/blob/3756e368c83bf68c61a983263a6cd081f9864244/lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_order.rb#L310

Now looking into the [changes of version 3.1](https://github.com/ruby/bigdecimal/blob/master/CHANGES.md#310) I found several entries of fixes for `x.to_d * y.to_f` and `x.to_d / y.to_f`. Those had rounding errors before but now they seem correct.

* https://github.com/ruby/bigdecimal/issues/70

So this change makes sure that we get to the correct result. But I think that this is another example for providing the literal value in test cases:

```suggestion
        # (113.3 - 10) - (3.3 - 0.29)
        expect(total).to eq 100.29
```

Before we got the wrong total and compared it to the wrong expected result:

```
# We were testing for this value:
(113.3 - 10) - (3.3 - 0.29)
=> 100.28999999999999

# Now we are testing for the correct value:
(BigDecimal('113.3') - 10) - (BigDecimal('3.3') - BigDecimal('0.29'))
=> 0.10029e3
```

Reply:
> this is another example for providing the literal value in test cases

So true! 👍 👍